packages/client/src/generation/generateClient.ts

```diff
@@ -250,10 +257,10 @@ export async function generateClient(options: GenerateClientOptions): Promise<vo
       const filePath = path.join(finalOutputDir, fileName)
       // The deletion of the file is necessary, so VSCode
       // picks up the changes.
-      if (await exists(filePath)) {
-        await fs.promises.unlink(filePath)
+      if (existsSync(filePath)) {
+        await fs.unlink(filePath)
```

Comment:
We could unlink the file without checking and ignore the error if it doesn't exist. This is technically more correct in general case because doing this in two calls introduces a race condition as another process can remove the file in between, but I'm not sure if this argument stands here given that there are probably other things that could fail if another generator were to modify the directory concurrently. So feel free to ignore this if you think it doesn't apply here.

On the other hand, it could also be a tiny bit more efficient if the file is more likely to exist than not to.

Reply:
It's a valid point, but since this PR does not attempt to touch this, I suggest we improve it via another PR.